### PR TITLE
[1247] add transition info page

### PR DESCRIPTION
**Pull request by misaka**
### Context

New users should see the transition info page one time once they're transitioned.

### Changes proposed in this pull request

Display the transition info page to new users, and once viewed fire an event to the backend so that they don't see it again.

### CODE REVIEW

**Comment by dankmitchell** [comment]
Look good.

Im going to look at this final acceptance criteria:

- users who haven't signed the Ts & Cs don't need to see the transition screen

File: spec/features/view_pages_spec.rb [hunk 1]
```diff
@@ -30,11 +30,11 @@
     expect(find('h1')).to have_content('Guidance for Publish teacher training courses')
   end
 
-  scenario "Navigate to /transition" do
+  scenario "Navigate to /transition_info" do
```
[the remark this answers is not in the document]
Comment (misaka) [reply spec/features/view_pages_spec.rb]: I think you're right. I'll look into rolling this into another test.

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -7,9 +7,13 @@ def create
     session[:auth_user] = auth_hash
 
     add_token_to_connection
-    set_user_session
+    user = set_user_session
 
-    redirect_to root_path
+    if user.state == 'new'
```
[the remark this answers is not in the document]
Comment (benilovj) [reply app/controllers/sessions_controller.rb]: Never mind – this opened up a whole can'o'worms on the tests behaving differently to the real app code 🙀 .

File: spec/features/view_pages_spec.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (benilovj) [reply spec/features/view_pages_spec.rb]: Updated in d36be0d

**Comment by benilovj** [comment]
> * users who haven't signed the Ts & Cs don't need to see the transition screen

@dankmitchell this is now done in d36be0d

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class UsersController < ApplicationController
+  def accept_transition_info
+    user = User.new(id: current_user["user_id"])
```
Comment (misaka) [inline app/controllers/users_controller.rb]: Would it make sense to try to hide this little bit of ugly use of `new` by creating a method to mask it? 

```
User.member(current_user['user_id']).accept_transition_user

# and in User:

def self.member(id)
  # BAD JSON API client ... BAD BAD BAD JSON API client
  new(id)
end
```

?

Comment (benilovj) [reply app/controllers/users_controller.rb]: Sure - fixed in 272bcd9

**Comment by misaka** [comment]
LGTM. 👍 Would approve if I had the choice.

**Pull request merged**